mrp_sale_info/models/sale_order.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2018 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    production_ids = fields.One2many('mrp.production', 'sale_id')
+    production_count = fields.Integer(
+        compute='_compute_production_count', store=True)
+
+    @api.depends("production_ids")
+    def _compute_production_count(self):
+        for sale in self:
+            sale.production_count = len(sale.production_ids)
+
+    @api.multi
+    def action_view_production(self):
+        action = self.env.ref('mrp.mrp_production_action').read()[0]
+        if self.production_count > 1:
+            action['domain'] = [('id', 'in', self.production_ids.ids)]
+        else:
+            action['views'] = [
+                (self.env.ref('mrp.mrp_production_form_view').id, 'form')]
+            action['res_id'] = self.production_ids.id
+        return action
```

Comment:
For the sake of coverage, could you add a simple assert of `sale.production_count == 1` and a call to [action_view_production](https://github.com/OCA/manufacture/blob/12.0/mrp_sale_info/tests/test_mrp_sale_info.py#L39) in the tests here, asserting that the response in `res_id` is the production order we are expecting? It won't cover the `> 1` case, but it should already be a better coverage.